lib/maplibre-compose/src/maplibreNativeMain/kotlin/dev/sargunv/maplibrecompose/compose/offline/OfflineRegionManager.kt

```diff
@@ -0,0 +1,26 @@
+package dev.sargunv.maplibrecompose.compose.offline
+
+import androidx.compose.runtime.Composable
+
+@Composable public expect fun rememberOfflineRegionManager(): OfflineRegionManager
+
+public interface OfflineRegionManager {
+  public val regions: Set<OfflineRegion>
+
+  public suspend fun create(
+    definition: OfflineRegionDefinition,
+    metadata: ByteArray = ByteArray(0),
+  ): OfflineRegion
+
+  public suspend fun delete(region: OfflineRegion)
+
+  public suspend fun invalidate(region: OfflineRegion)
+
+  public suspend fun invalidateAmbientCache()
+
+  public suspend fun clearAmbientCache()
+
+  public suspend fun setMaximumAmbientCacheSize(size: Long)
```

Comment:
This would probably also need some explanation what an "ambient cache" is. First thing that comes up in a google search is MapBox documentation:

> When a map is loaded and viewed on a device, the tiles are placed in the device's ambient cache. The ambient cache stores tiles that have already been visited by an end user so that the next time the user returns, the map loads nearly instantaneously.

Reply:
Yup, it's the automatically cached offline tiles